doc/manual/syntax.md

```diff
@@ -0,0 +1,313 @@
+Elements inside a record are unordered.
+Two records can be _merged_ together using the operator `&`. The reader can find more information about merging in the relevant documentation.
+
+Examples:
+```
+{}
+{a = 3}
+{my_id_n5 = "my id number 5", my_id_n4 = }
+{"5" = 5, six = 6}
+```
+
+Accessing a record field can be done using the `.` operator :
+```
+> { a = 1, b = 5 }.a
+1
+
+> { a = 1 }.b
+error: Missing field
+
+> { "1" = "one" }."1"
+"one"
+```
+
+It is possible to write records of records via the *pieces syntax*, where we separate fields by dots:
+```
+> { a = { b = 1 } }
+{ a = { b = 1 } }
+ 
+> { a.b = 1 }
+{ a = { b = 1 } }
+
+> { a.b = 1, a.c = 2, b = 3}
+{ a = { b = 1, c = 2 }, b = 3 }
+```
+
+## Statements
+
+### If-Then-Else
+This statement allows conditional branching in your code. You can use it as `if <bool expr> then <expr> else <expr>`. Both expressions in the `then` and `else` branches are lazily evaluated, and must have the same type.
+
+Examples:
+```
+> if true then "TRUE :)" else "false :("
+"TRUE :)"
+
+> if false then "Not this one" else "This one"
+"This one"
+
+> if "forty-two" == 42 then "equal?" else "unequal"
+"unequal"
+```
+
+### Let-In
+Let-in allows the binding of an expression. It is used as `let <ident> = <expr> in <expr>`.
+
+Examples:
+```
+> let r = { a = "a", b = "b" } in r.a
+"a"
+
+> let inner = { inside = true } in let outer = { outside = inner.inside } in outer.outside
+true
+
+> let a = 1 in let b = 2 in a + b
+3
+```
+
+## Functions
+A function is declared using the `fun` keyword, then arguments separated by spaces, and finally an arrow `=>` to add the body of the function.
+To call a function, just add the arguments after it separated with spaces.
+Functions in Nickel are curried, meaning that a function taking multiple arguments is actually a function that takes a single argument and returns a function taking the rest of the arguments, until it is applied.
+
+Example
+```
+> (fun a b => a + b) 1 2
+3
+
+let add = fun a b => a + b in add 1 2
+3
+
+> let add = fun a b => a + b in
+    let add1 = add 1 in
+        add1 2
+3
+```
+
+## Typing
+To give a type to a value, we write it with `< value > : < type >`.
+More information on typing in the relevant document.
+
+Examples:
+```
+5 : Num
+"Hello" : Str
+
+(fun a b => a + b) : Num -> Num -> Num
+let add : Num -> Num -> Num = fun a b => a + b
+
+{a: Num = 1, b: Bool = true, c : List Num = [ 1 ]}
+let r : {a : Num, b : Bool, c : List Num} = { a = 1, b = true, c = [ 1 ] }
+
+{ a = 1, b = 2 } : { _ : Num }
+let r : { _ : Num } = { a = 1, b = 2 }
+```
+
+## Metadata
+Metadata can be used to add contracts (more information in relevant documentation), documentation or default values.
+It is introduced with a `|`. Multiple metadata can be chained.
+
+For adding a regular type as contract, the syntax is `| < type >`. For a defined contract, we use the `| #< defined type >`.
```

Comment:
```suggestion
```
I think we can just get rid of this one, since this won't be true anymore once we have merged the type and contract syntax.